src/v/cloud_storage/remote_segment.cc

```diff
@@ -883,22 +904,58 @@ ss::future<> remote_segment::hydrate() {
       .discard_result();
 }
 
-ss::future<> remote_segment::hydrate_chunk(
-  chunk_start_offset_t start, std::optional<chunk_start_offset_t> end) {
+remote_segment::consume_stream::consume_stream(
+  remote_segment& remote_segment, segment_chunk_range range)
+  : _segment{remote_segment}
+  , _range{std::move(range)} {}
+
+ss::future<uint64_t> remote_segment::consume_stream::operator()(
+  uint64_t size, ss::input_stream<char> stream) {
+    for (const auto [start, end] : _range) {
+        auto dsi = std::make_unique<bounded_stream>(
+          stream, end.value_or(_segment._size));
+        auto stream_upto = ss::input_stream<char>{
+          ss::data_source{std::move(dsi)}};
+        co_await _segment.put_chunk_in_cache(
+          size, std::move(stream_upto), start);
+    }
+    co_return size;
+}
+
+ss::future<> remote_segment::hydrate_chunk(segment_chunk_range range) {
+    const auto start = range.first_offset();
+    const auto path_to_start = get_path_to_chunk(start);
+
+    // It is possible that the chunk has already been downloaded during a
+    // prefetch operation. In this case we skip hydration and try to materialize
+    // the chunk. This also skips the prefetch of the successive chunks. So
+    // given a series of chunks A, B, C, D, E and a prefetch of 2, when A is
+    // fetched B,C are also fetched. Then hydration of B,C are no-ops and no
+    // prefetch is done during those no-ops. When D is fetched, hydration
+    // makes an HTTP GET call and E is also prefetched. So a total of two calls
+    // are made for the five chunks (ignoring any cache evictions during the
+    // process).
+    if (const auto status = co_await _cache.is_cached(path_to_start);
```

Comment:
Each `is_cached` call maps to an `access` sys call. We should be able to answer these queries by using the access time tracker in `cache_service`. I'll try to include this in https://github.com/redpanda-data/redpanda/pull/10855.